packages/analytics-v1.1/src/core/analytics.js

```diff
@@ -587,19 +591,23 @@ class Analytics {
     //   object.clientIntegrationObjects.length
     // );
 
-    if (object.clientIntegrationObjects.every(intg => !intg.isReady || intg.isReady())) {
-      // Integrations are ready
-      // set clientIntegrationsReady to be true
-      this.integrationsData = constructMessageIntegrationsObj(
-        this.integrationsData,
-        object.clientIntegrationObjects,
-      );
-      object.clientIntegrationsReady = true;
-      // Execute the callbacks if any
-      object.executeReadyCallback();
-    }
+    try {
+      if (object.clientIntegrationObjects.every(intg => !intg.isReady || intg.isReady())) {
+        // Integrations are ready
+        // set clientIntegrationsReady to be true
+        this.integrationsData = constructMessageIntegrationsObj(
+          this.integrationsData,
+          object.clientIntegrationObjects,
+        );
+        object.clientIntegrationsReady = true;
+        // Execute the callbacks if any
+        object.executeReadyCallback();
+      }
 
-    this.processBufferedCloudModeEvents();
+      this.processBufferedCloudModeEvents();
```

Comment:
Let's keep this outside of this try-catch. Failure to ensure ready status should not hamper the data plane events delivery.